=================================================

Milestone

Nama  : Gracia Valerine

Batch : HCK-017

Program ini dibuat untuk menyelesaikan masalah business untuk menaikkan potensi pendapatan.


=================================================

Link Dashboard Tableau: https://public.tableau.com/shared/JPFC3SSQ6?:display_count=n&:origin=viz_share_link

    A. IDENTIFIKASI MASALAH

LATAR BELAKANG:

Strategi pemasaran efektif sangatlah penting untuk meningkatkan penjualan dalam persaingan bisnis. Dari data yang didapatkan, ditunjukkan variasi perilaku pembelian berdasarkan jenis kelamin dan lokasi geografis yang dapat membantu menghadapi tantangan untuk mentapai target penjualan, terutama produk yang minim dibeli.

Program ini dibentuk dengan tujuan untuk mengidentifikasi produk yang jarang dibeli, memahami pola pembelian berdasarkan jenis kelamin, dan menganalisis distribusi lokasi pembelian. Informasi tersebut dapat membantu menyusun strategi pemasaran agar tepat sasaran untuk meningkatkan penjualan produk tersebut dan mencapai peningkatan pendapatan penjualan sebesar 10% dalam 3 bulan.

Analisis data ini mencakup:
* Identifikasi produk dengan penjualan minim
* Pengelompokan pelanggan berdasarkan jenis kelamin
* Analisis korelasi antara produk dan jenis pelanggan
* Identifikasi kota dengan jumlah pelanggan minim
* Analisis pola penjualan musiman

Penerapan hasil analisis ke dalam kampanye pemasaran spesifik dan terarah ditujukan untuk mendapat peningkatan penjualan yang signifikan. Implementasi strategi ini diharapkan akan efektif untuk mendapat peningkatan revenue sebesar 10% dalam periode 3 bulan. 

TOPIK PERMASALAHAN:

Mengoptimalkan Marketing dengan Menjangkau Customer Secara Spesifik untuk Menaikkan Sales Revenue

SMART METHOD:

S - mengoptimalkan marketing dengan mejangkau customer secara spesifik 

M -  Menaikkan sales revenue sebesar 10% 

A -Analisa data produk yang jarang dibeli, jenis kelamin pelanggan mana yang jarang membeli produk, dan korelasi produk dengan jenis pelanggan 

R - Menargetkan pelanggan secara spesifik dapat menaikkan pendapatan penjualan secara tidak langsung 

T - Rencana ditargetkan untuk 3 bulan kedepan

PROBLEM STATEMENT:
 
Mengoptimalkan marketing dengan mejangkau customer secara specifik untuk menaikkan sales revenue sebesar 10% dengan analisa data produk yang jarang dibeli, jenis kelamin pelanggan mana yang jarang membeli produk, dan korelasi produk dengan jenis pelanggan dalam 3 bulan kedepan.

5W + 1H:

1. Produk apa yang paling minim dibeli? (Visualisasi pie chart)
2. Pelanggan dengan jenis kelamin apa yang jarang membeli produk? (Visualisasi bar chart)
3. Apa & bagaimana korelasi produk dengan jenis kelamin pelanggan? (S. Deskriptif)
4. Kota mana yang paling minim pelanggan? (Visualisasi pie chart)
5. Bulan apa yang minim pelanggan? (Visualisasi bar chart)
6. Apa & bagaimana korelasi kota dengan bulan pembelian? (S. Deskriptif)
7. kenapa perlu mengerahkan marketing untuk menaikkan revenue 10% ke produk yg jarang dibeli? (S. Inferensial)

    B. DATA LOADING

In [16]:
#Import package yang akan digunakan dalam program
import time
import pandas as pd
from scipy import stats
import math
import numpy as np
from scipy.stats import pearsonr, ttest_rel
import matplotlib.pyplot as plt
import plotly.express as px
from datetime import datetime

#Memasukkan file csv untuk dibaca sebagai dataframe
supermarket = pd.read_csv('supermarket_sales - Sheet1.csv')

In [17]:
#Menampilkan sample isi data secara acak
supermarket.sample(10)

Invoice ID Branch       City Customer type  Gender  \
352  115-99-4379      B   Mandalay        Member  Female   
906  450-42-3339      C  Naypyitaw        Normal    Male   
377  743-88-1662      C  Naypyitaw        Normal    Male   
123  336-78-2147      C  Naypyitaw        Member    Male   
532  857-67-9057      B   Mandalay        Normal    Male   
991  602-16-6955      B   Mandalay        Normal  Female   
802  474-33-8305      C  Naypyitaw        Member    Male   
765  593-14-4239      B   Mandalay        Normal  Female   
463  320-32-8842      C  Naypyitaw        Member  Female   
713  268-20-3585      C  Naypyitaw        Normal  Female   

               Product line  Unit price  Quantity   Tax 5%     Total  \
352     Fashion accessories       54.73         7  19.1555  402.2655   
906       Health and beauty       84.61        10  42.3050  888.4050   
377       Sports and travel       95.49         7  33.4215  701.8515   
123      Home and lifestyle       63.91         8  25.5640  536.8440   
532  Electronic accessories       22.95        10  11.4750  240.9750   
991       Sports and travel       76.60        10  38.3000  804.3000   
802     Fashion accessories       67.39         7  23.5865  495.3165   
765      Home and lifestyle       95.46         8  38.1840  801.8640   
463      Food and beverages       22.62         1   1.1310   23.7510   
713       Health and beauty       13.85         9   6.2325  130.8825   

          Date   Time      Payment    cogs  gross margin percentage  \
352  3/14/2019  19:02  Credit card  383.11                 4.761905   
906   2/9/2019  18:58  Credit card  846.10                 4.761905   
377  2/22/2019  18:17      Ewallet  668.43                 4.761905   
123  3/13/2019  19:52  Credit card  511.28                 4.761905   
532   2/6/2019  19:20      Ewallet  229.50                 4.761905   
991  1/24/2019  18:10      Ewallet  766.00                 4.761905   
802  3/23/2019  13:23      Ewallet  471.73                 4.761905   
765   3/5/2019  19:40      Ewallet  763.68                 4.761905   
463  3/17/2019  18:58         Cash   22.62                 4.761905   
713   2/4/2019  12:50      Ewallet  124.65                 4.761905   

     gross income  Rating  
352       19.1555     8.5  
906       42.3050     8.8  
377       33.4215     8.7  
123       25.5640     4.6  
532       11.4750     8.2  
991       38.3000     6.0  
802       23.5865     6.9  
765       38.1840     4.7  
463        1.1310     6.4  
713        6.2325     6.0

In [18]:
#Menampilkan informasi terkait data yang dimasukkan (jumblah data dan jenis data)
supermarket.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

Insight: Dari data yang di atas, hampir semua data tersimpan dalam format yang benar. Tanggal dan waktu saja yang perlu diubah menjadi format 'date time'.

In [19]:
supermarket.isnull()

Invoice ID  Branch   City  Customer type  Gender  Product line  \
0         False   False  False          False   False         False   
1         False   False  False          False   False         False   
2         False   False  False          False   False         False   
3         False   False  False          False   False         False   
4         False   False  False          False   False         False   
..          ...     ...    ...            ...     ...           ...   
995       False   False  False          False   False         False   
996       False   False  False          False   False         False   
997       False   False  False          False   False         False   
998       False   False  False          False   False         False   
999       False   False  False          False   False         False   

     Unit price  Quantity  Tax 5%  Total   Date   Time  Payment   cogs  \
0         False     False   False  False  False  False    False  False   
1         False     False   False  False  False  False    False  False   
2         False     False   False  False  False  False    False  False   
3         False     False   False  False  False  False    False  False   
4         False     False   False  False  False  False    False  False   
..          ...       ...     ...    ...    ...    ...      ...    ...   
995       False     False   False  False  False  False    False  False   
996       False     False   False  False  False  False    False  False   
997       False     False   False  False  False  False    False  False   
998       False     False   False  False  False  False    False  False   
999       False     False   False  False  False  False    False  False   

     gross margin percentage  gross income  Rating  
0                      False         False   False  
1                      False         False   False  
2                      False         False   False  
3                      False         False   False  
4                      False         False   False  
..                       ...           ...     ...  
995                    False         False   False  
996                    False         False   False  
997                    False         False   False  
998                    False         False   False  
999                    False         False   False  

[1000 rows x 17 columns]

Insight: Dari data yang di atas, tidak ada data yang kosong.

    C. DATA CLEANING

In [20]:
#Mengubah data tanggal dari format string menjadi format tanggal
supermarket['Date'] = pd.to_datetime(supermarket['Date'], infer_datetime_format=True, dayfirst=False)

/var/folders/xr/3l48bz352q5cfyjkqwysr4v40000gp/T/ipykernel_61876/219670442.py:2: UserWarning:

The argument 'infer_datetime_format' is deprecated and will be removed in a future version. A strict version of it is now the default, see https://pandas.pydata.org/pdeps/0004-consistent-to-datetime-parsing.html. You can safely remove this argument.



In [21]:
#Mengubah data jam dari string menjadi format jam
timeFormat  = "%H:%M"
supermarket['Time'] = pd.to_datetime(supermarket['Time'], format=timeFormat)

In [22]:
#Menambah kolom untuk melihat bulan
supermarket['Month'] = supermarket['Date'].dt.month

In [23]:
#Menambah kolom total penjualan dari bulan
salesByMonth = supermarket.groupby('Month')['Total'].sum()
supermarket['Monthly sales'] = supermarket['Month'].map(salesByMonth)

In [24]:
#Melihat data kembali
supermarket.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715 2019-01-05   
1  Electronic accessories       15.28         5   3.8200   80.2200 2019-03-08   
2      Home and lifestyle       46.33         7  16.2155  340.5255 2019-03-03   
3       Health and beauty       58.22         8  23.2880  489.0480 2019-01-27   
4       Sports and travel       86.31         7  30.2085  634.3785 2019-02-08   

                 Time      Payment    cogs  gross margin percentage  \
0 1900-01-01 13:08:00      Ewallet  522.83                 4.761905   
1 1900-01-01 10:29:00         Cash   76.40                 4.761905   
2 1900-01-01 13:23:00  Credit card  324.31                 4.761905   
3 1900-01-01 20:33:00      Ewallet  465.76                 4.761905   
4 1900-01-01 10:37:00      Ewallet  604.17                 4.761905   

   gross income  Rating  Month  Monthly sales  
0       26.1415     9.1      1     116291.868  
1        3.8200     9.6      3     109455.507  
2       16.2155     7.4      3     109455.507  
3       23.2880     8.4      1     116291.868  
4       30.2085     5.3      2      97219.374

    D. ANALISIS DAN PERHITUNGAN

1. Produk yang jarang dibeli

In [25]:
# Membuat visualisasi pie chart dari frekuensi produk
fig = px.pie(supermarket, names='Product line')
fig.update_layout(title_x=0.5)
fig.show()

Insight: Produk 'Health and beauty' paling jarang dibeli.

2. Jenis kelamin yang lebih jarang belanja

In [26]:
# Membuat visualisasi bar chart dari frekuensi jenis kelamin
fig = px.bar(customerCounts, x=customerCounts.index, y=customerCounts.values,
             labels={'x': 'Gender', 'y': 'Count'},
             title='Frekuensi Gender Pembeli')

fig.show()

Insight: Dari data yang diatas, laki-laki lebih jarang berbelanja dibanding wanita.

3. Korelasi produk dengan jenis kelamin pelanggan

In [29]:
#Memudahkan perhitungan korelasi dengan membuat kolom baru
productCounts = supermarket['Product line'].value_counts()
customerCounts = supermarket['Gender'].value_counts()
supermarket['Customer counts'] = supermarket['Gender'].map(customerCounts)
supermarket['Product count'] = supermarket['Product line'].map(productCounts)

In [30]:
#Mencari korelasi antara produk dengan jenis kelamin pelanggan
corr, p_value = pearsonr(supermarket['Product count'], supermarket['Customer counts'])
print(f'Correlation: {corr:.2f}, p value: {p_value:.2f}')

Correlation: 0.06, p value: 0.04


Insight: Koefesien korelasi antara jenis produk dan jenis kelamin pelanggan adalah 0.06 dan memiliki p value 0.04.

4. Kota dengan pelanggan paling sedikit

In [33]:
#Visualisasi data frekuensi pembelian berdasarkan kota
fig = px.pie(supermarket, names='City')
fig.update_layout(title_x=0.5)
fig.show()

Insight: Kota Mandalay paling jarang berbelanja dibanding kota lain.

5. Bulan dengan pembelian paling minim

In [32]:
#Menampilkan grafik garis untuk penjualan per bulan
fig = px.bar(supermarket,
              x='Month',
              y='Monthly sales',
              title='Monthly Sales')

fig.show()

Insight: Transaksi dalam bulan Februari paling sedikit.

6. Korelasi antara bulan dan kota 

In [34]:
#Mencari korelasi antara kota dengan bulan penjualan rendah
cityCounts = supermarket['City'].value_counts()
monthCounts = supermarket['Month'].value_counts()
corr, p_value = pearsonr(monthCounts, cityCounts)
print(f'Correlation: {corr:.2f}, p value: {p_value:.2f}')

Correlation: 0.84, p value: 0.37


Insight: Koefisien korelasi kota dan bulan penjualan adalah 0.84 dan p value-nya adalah 0.37.

7. Alasan untuk mengerahkan marketing untuk menaikkan revenue 10% ke produk yg jarang dibeli

In [35]:
#Rata-rata pendapatan
avgPendapatanLama = supermarket['Unit price'] * supermarket['Quantity'].mean()
avgPendapatanBaru = avgPendapatanLama * 1.10 

#Hypothesis testing
h0 = "Rata-rata pendapatan setelah menerapkan strategi baru kurang dari atau sama dengan rata-rata pendapatan sekarang"
h1 = "Rata-rata pendapatan setelah menerapkan strategi baru lebih dari rata-rata pendapatan sekarang"

#T-test
t_stat, p_value = stats.ttest_ind(avgPendapatanBaru, avgPendapatanLama, alternative='greater')

print(f'Test Statistic: {t_stat:.2f}')
print(f'p-value: {p_value:.4f}')

#Hasil dari hipotesis
alpha = 0.05
if p_value < alpha:
    print("H0 ditolak, ada kenaikkan signifikan dalam pendapatan.")
else:
    print("H0 gagal ditolak, tidak ada kenaikkan signifikan dalam pendapatan.")


Test Statistic: 4.47
p-value: 0.0000
H0 ditolak, ada kenaikkan signifikan dalam pendapatan.


Insight: T-test memiliki nilai 4.47 dan p-value memiliki nilai 0, sehingga h0 ditolak, ada kenaikkan signifikan dalam pendapatan.

In [ ]:
#Convert data frame menjadi file excel
supermarket.to_excel('supermarket_sales.xlsx')  

    E. PENGAMBILAN KESIMPULAN

Berdasarkan data yang ada di atas, produk 'Health and beauty' adalah produk yang paling jarang dibeli. Hal ini menandakan bahwa produk tersebut kurang diminati atau kurang dipromosikan dengan baik.

Berdasarkan jenis kelamin, laki-laki cenderung lebih jarang berbelanja dibanding wanita, hal ini dapat menandakan bahwa target promosi kurang diarahkan kepada calon pelanggan laki-laki.

Jenis produk dan jenis kelamin pelangan tidak memiliki hubungan yang signifikan. Sehingga, cakupan target promosi setiap produk dapat diperluas, tidak hanya terarah pada satu jenis kelamin saja. 

Kota Mandalay merupakan kota yang memilki frekuensi pembelian terendah. Hal ini menunjukkan bahwa mungkin diperlukan pendekatan pemasaran yang berbeda atau peningkatan untuk meningkatkan jumlah pembelian.

Bulan Februari memiliki frekuensi pembelian yang rendah dalam 3 bulan. Hal ini menunjukkan bahwa marketing atau promosi produk dalam bulan tersebut kurang efektif.

Kota dan bulan pembelian memiliki korelasi yang lumayan signifikan. Hal tersebut mengartikan bahwa jumblah penjualan di kota dipengaruhi waktu dalam bulan.

Dengan meningkatkan pemasaran pada produk yang jarang dibeli, perusahaan dapat memanfaatkan potensi pasar yang belum efektif sepenuhnya. Dengan menaikkan revenue 10%, ini akan membantu memperbaiki profitabilitas produk tersebut dan meningkatkan total revenue perusahaan. Setelah implementasi strategi marketing baru, akan ada kenaikkan pendapatan secara signifikan jika berjalan dengan baik.

    F. SARAN

Produk "Health and beauty" dapat dipromosikan lebih sering lagi, terlebih pada pelanggan laki-laki di kota Naypyitaw pada bulan Februari. Karena pembelian jenis produk tidak berhubungan erat dengan jenis kelamin pembeli, maka tim marketing dapat memperkenalkan produk-produk "Health and beauty" pada kalangan laki-laki. Kota Mandalay memerlukan promosi lebih efektif pada bulan Februari, karena penjualan kota berhubungan erat dengan bulan.

Dengan melakukan inplementasi strategi marketing baru, maka diharapkan bahwa akan ada kenaikkan pendapatan secara signifikan.

    G. LAMPIRAN

Link Dashboard Tableau: https://public.tableau.com/shared/JPFC3SSQ6?:display_count=n&:origin=viz_share_link